relay-general/src/store/normalize/contexts.rs

```diff
@@ -199,12 +204,44 @@ fn normalize_response(response: &mut ResponseContext) {
     }
 }
 
-pub fn normalize_context(context: &mut Context) {
+static ANDROID_MAP: OnceCell<Result<HashMap<String, String>, std::io::Error>> = OnceCell::new();
+
+fn get_android_product_name(model: &str) -> Option<String> {
+    let mymap: &Result<HashMap<String, String>, std::io::Error> = ANDROID_MAP.get_or_init(|| {
+        let path = PathBuf::from("../android_models.csv");
+        let file = std::fs::File::open(&path)?;
+        let reader = std::io::BufReader::new(file);
+        let mut android_map: HashMap<String, String> = HashMap::new();
+
+        for line in reader.lines() {
+            let line = line?;
+            let columns: Vec<&str> = line.split(',').collect();
+
+            if columns.len() >= 4 {
+                let key = columns[3].trim().to_string();
+                let value = columns[1].trim().to_string();
+                android_map.insert(key, value);
+            }
+        }
```

Comment:
Adding another dependency to deal with a single CSV file may not be worth it, but I don't like parsing it manually for the number of things that can go wrong. See, for example, [this StackOverflow](https://stackoverflow.com/a/10141147) answer about the delimiter (`,` or `;`).

I guess it's fine to hardcode it like this since we're inserting the file at compile time.

Reply:
are you thinking of this crate?: https://crates.io/crates/csv

now we are back to run-time parsing so this is more relevant again.